core/amber/src/main/scala/edu/uci/ics/texera/web/resource/GmailResource.scala

```diff
@@ -0,0 +1,122 @@
+import com.google.api.services.gmail.model.Message
+import com.google.auth.http.HttpCredentialsAdapter
+import com.google.auth.oauth2.{AccessToken, GoogleCredentials}
+import edu.uci.ics.amber.engine.common.AmberUtils
+import edu.uci.ics.texera.Utils
+import edu.uci.ics.texera.web.auth.SessionUser
+import io.dropwizard.auth.Auth
+import org.apache.commons.codec.binary.Base64
+
+import java.io.ByteArrayOutputStream
+import java.nio.file.Files
+import java.util.{Collections, Properties}
+import javax.annotation.security.RolesAllowed
+import javax.mail.Session
+import javax.mail.internet.{InternetAddress, MimeMessage}
+import javax.ws.rs._
+
+case class EmailMessage(email: String, subject: String, content: String)
+@Path("/gmail")
+class GmailResource {
+  final private lazy val path =
+    Utils.amberHomePath.resolve("src").resolve("main").resolve("resources").resolve("gmail")
+  final private lazy val clientId = AmberUtils.amberConfig.getString("user-sys.google.clientId")
+  final private lazy val clientSecret =
+    AmberUtils.amberConfig.getString("user-sys.google.clientSecret")
+
+  @GET
+  @RolesAllowed(Array("ADMIN"))
+  @Path("/email")
+  def getEmail: String = {
+    new String(Files.readAllBytes(path.resolve("userEmail")))
+  }
+
+  @POST
+  @RolesAllowed(Array("ADMIN"))
+  @Path("/auth")
+  def saveRefreshToken(code: String): Unit = {
+    this.deleteEmail()
+    val token = new GoogleAuthorizationCodeTokenRequest(
+      new NetHttpTransport(),
+      new GsonFactory(),
+      clientId,
+      clientSecret,
+      code,
+      "postmessage"
+    ).execute()
+
+    Files.write(
+      path.resolve("refreshToken"),
+      token.getRefreshToken.getBytes
+    )
+
+    Files.write(
+      path.resolve("userEmail"),
+      new GoogleIdTokenVerifier.Builder(new NetHttpTransport, GsonFactory.getDefaultInstance)
+        .setAudience(Collections.singletonList(clientId))
+        .build()
+        .verify(token.getIdToken)
+        .getPayload
+        .getEmail
+        .getBytes
+    )
+  }
+
+  @DELETE
+  @RolesAllowed(Array("ADMIN"))
+  @Path("/revoke")
+  def deleteEmail(): Unit = {
+    if (Files.exists(path.resolve("userEmail"))) Files.delete(path.resolve("userEmail"))
+    if (Files.exists(path.resolve("refreshToken"))) Files.delete(path.resolve("refreshToken"))
+    if (Files.exists(path.resolve("accessToken"))) Files.delete(path.resolve("accessToken"))
+  }
+
+  @PUT
+  @RolesAllowed(Array("REGULAR", "ADMIN"))
+  @Path("/send")
+  def sendEmail(emailMessage: EmailMessage, @Auth user: SessionUser): Unit = {
+    val accessTokenPath = path.resolve("accessToken")
+    if (!Files.exists(accessTokenPath)) {
+      Files.write(
+        accessTokenPath,
+        new GoogleRefreshTokenRequest(
+          new NetHttpTransport(),
+          new GsonFactory(),
+          new String(Files.readAllBytes(path.resolve("refreshToken"))),
+          clientId,
+          clientSecret
+        ).execute().getAccessToken.getBytes
```

Comment:
I think you need to add some error handling here, unless you are sure that this `GoogleRefreshTokenRequest` will always be successful (which I don't think is the case). Also it would be good to check if the necessary tokens/files really exist before executing the send.